### PR TITLE
Move some model classes to Protobuf classes [2/?]

### PR DESCRIPTION
**Pull request by prm-james-hill**

### CODE REVIEW

File: src/main/java/ai/promoted/delivery/client/InsertionFactory.java [hunk 1]
```diff
@@ -16,10 +19,50 @@ public class InsertionFactory {
...
+      for (Object item : iterable) {
+        listBuilder.addValues(convertObjectToValue(item));
+      }
+      return com.google.protobuf.util.Values.of(listBuilder.build());
+    } else if (object instanceof Map) {
+      @SuppressWarnings("unchecked")
+      Map<String, Object> map = (Map<String, Object>) object;
+      return com.google.protobuf.util.Values.of(convertMapToStruct(map));
+    }
+    return com.google.protobuf.Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build();
```
Comment (prm-james-hill) [inline src/main/java/ai/promoted/delivery/client/InsertionFactory.java:63]: @prm-dan Do you have a preference on how this is handled? Like if this should be logging an error, throwing some kind of unimplemented exception (and if so, do we catch it inside the SDK or document it in the README), etc.

Comment (prm-james-hill) [reply src/main/java/ai/promoted/delivery/client/InsertionFactory.java:63]: Actually, this isn't even used in the SDK (outside of test). So whatever we do here could be worth documenting

Comment (prm-dan) [reply src/main/java/ai/promoted/delivery/client/InsertionFactory.java:63]: Hmm, I see.  I'd probably throw if it's an unsupported type.  They'd hit it when developing.  You'll also want an explicit null check.

```
} else if (object == null) {
  return com.google.protobuf.Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build();
} else {
  throw new UnsupportedOperationException("property value class=" + object.getClass());
}
```

File: src/main/java/ai/promoted/delivery/client/InsertionFactory.java [hunk 2]
```diff
@@ -16,10 +19,50 @@ public class InsertionFactory {
...
+    } else if (object instanceof String) {
+      return com.google.protobuf.util.Values.of((String) object);
+    } else if (object instanceof Iterable) {
+      Iterable<?> iterable = (Iterable<?>) object;
+      ListValue.Builder listBuilder = ListValue.newBuilder();
+      for (Object item : iterable) {
+        listBuilder.addValues(convertObjectToValue(item));
+      }
+      return com.google.protobuf.util.Values.of(listBuilder.build());
+    } else if (object instanceof Map) {
```
Comment (prm-dan) [inline src/main/java/ai/promoted/delivery/client/InsertionFactory.java:58]: I'd put Map above Iterable.  I know Map doesn't extend Iterable but it made me do a double take.

**Review by prm-dan — APPROVED** [review]

File: src/main/java/ai/promoted/delivery/client/InsertionFactory.java (hunk 1 above)
Comment (prm-james-hill) [reply src/main/java/ai/promoted/delivery/client/InsertionFactory.java]: Done.

File: src/main/java/ai/promoted/delivery/client/InsertionFactory.java (hunk 2 above)
Comment (prm-james-hill) [reply src/main/java/ai/promoted/delivery/client/InsertionFactory.java]: Done.

**Pull request merged**